agents_infra/agents_infra/agents/base.py

```diff
@@ -156,6 +149,36 @@ def setup_logging(self, log_path=None):
             },
         )
 
+    def aggregate_reports(self, category=None):
+        """
+        Aggregate reports for a given category.
+
+        Parameters:
+            category (str, optional): Report category.
+        """
+        reports = {'server_name': self.server_name}
+
+        # Update report with server's identity data.
+        if category == 'status':
+            reports.update({'hostname': self.hostname, 'ip': self.ip})
+
+        for name, p in inspect.getmembers(
+            type(self),
+            predicate=lambda p: (
+                inspect.ismethod(p) and getattr(p, 'is_plugin', False)
+            )
+        ):
+            if p.enabled and (category is None or p.category == category):
```

Comment:
Does `enabled` always exist on the plugin method? Consider wrapping with `getattr(p, 'enabled', True)` to avoid `AttributeError`.

Reply:
It does always exist, yes, and is initialized in the constructor of `Plugin`.